Storage/SQL/RemoteDevicesTable.swift

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Shared
+
+public struct RemoteDevice {
+    public let id: String
+    public let name: String
+    public let type: String
+    public let isCurrentDevice: Bool
+    public let lastAccessTime: Timestamp?
+
+    public init(id: String, name: String, type: String, isCurrentDevice: Bool, lastAccessTime: Timestamp?) {
+        self.id = id
+        self.name = name
+        self.type = type
+        self.isCurrentDevice = isCurrentDevice
+        self.lastAccessTime = lastAccessTime
+    }
+}
+
+let TableRemoteDevices = "remote_devices"
+
+class RemoteDevicesTable<T>: GenericTable<RemoteDevice> {
```

Comment:
We are actively working to kill this pattern. Put this in `BrowserTable` with a migration instead. See Bug 1388147 and talk to @justindarc.